services/diversity/src/services/richness.js

```diff
@@ -12,50 +12,45 @@ module.exports = (RichnessPersistence, restAPI) => {
      *
      * @returns {Object[]} Number of inferred and observed species for the desired group.
      */
-    getNumberSpecies: async (areaType, areaId, group = 'all') => {
-      const data = [
-        {
-          id: 'total',
-          inferred: 40,
-          observed: 4,
-          region_observed: 110,
-          region_inferred: 130,
-        },
-        {
-          id: 'endemic',
-          inferred: 20,
-          observed: 25,
-          region_observed: 120,
-          region_inferred: 100,
-        },
-        {
-          id: 'invasive',
-          inferred: 20,
-          observed: 10,
-          region_observed: 90,
-          region_inferred: 80,
-        },
-        {
-          id: 'threatened',
-          inferred: 15,
-          observed: 20,
-          region_observed: 90,
-          region_inferred: 100,
-        },
-      ];
+    getNumberOfSpecies: async (areaType, areaId, group = 'all') => {
```

Comment:
There are some issues with this function:
- `values` isn't needed in the function scope, you can declare it in each case
- `result` will always have a value so the if in line 49 will never be used. And even if there are no results from the persistence in every case you are always adding something to `result`
- `result` doesn't need to be declared with let (`push` isn't detected as a modification by eslint, and I don't know what the heck is happening with the aribnb-base rules)
- `lab` is only used in the `'all'` case doesn't need to be in the function scope
- Every time you use `await` you need to put it inside a `try - catch` statement
- The default case is returning an empty array, but as the `group` param has a default value of `'all'` it will never be used.

I think in a couple of options to improve this implementation:

1. Let the `values` variable function scope and save there only the result of the promises (every case should use Promise.all so it always return an array). Encapsulate the whole switch case in the `try - catch`
2. Declare a variable to store an array of promises, in every case save only the promise (no `await`) then do a Promise.all inside a `try - catch` or with .then().catch()

In all cases you should return a message when there is an error.